packages/std/src/imports.rs

```diff
@@ -171,15 +176,20 @@ impl Api for ExternalApi {
         let send_ptr = &*send as *const Region as u32;
         let human = alloc(HUMAN_ADDRESS_BUFFER_LENGTH);
 
-        let read = unsafe { humanize_address(send_ptr, human as u32) };
-        if read < 0 {
-            return Err(StdError::generic_err("humanize_address returned error"));
+        let result = unsafe { humanize_address(send_ptr, human as u32) };
+        if result != 0 {
+            let error_data = unsafe { consume_region(result as *mut Region) };
+            let error = String::from_utf8_lossy(&error_data);
+            return Err(StdError::generic_err(format!(
+                "humanize_address errored: {}",
+                error
+            )));
         }
 
         let out = unsafe { consume_region(human) };
-        // we know input was correct when created, so let's save some bytes
-        let result = unsafe { String::from_utf8_unchecked(out) };
-        Ok(HumanAddr(result))
+        // We trust the VM/chain to return correct UTF-8, so let's save some gas
+        let out_string = unsafe { String::from_utf8_unchecked(out) };
+        Ok(HumanAddr(out_string))
```

Comment:
Similarly to this comment, can we not expect the errors to be valid utf-8? the VM can verify that the FFI returned valid data before giving it to the contract. The difference isn't that big, it's just a little bit less code on the contract side.

Reply:
Implemented in a1e2e1d